Microsoft.Azure.Cosmos/src/FeedRangeIteratorCore.cs

```
@@ -0,0 +1,236 @@
+//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos.CosmosElements;
+    using Microsoft.Azure.Cosmos.Query.Core;
+    using Microsoft.Azure.Cosmos.Query.Core.Monads;
+    using Microsoft.Azure.Cosmos.Resource.CosmosExceptions;
+    using Microsoft.Azure.Documents;
+
+    /// <summary>
+    /// Cosmos feed stream iterator. This is used to get the query responses with a Stream content
+    /// </summary>
+    internal sealed class FeedRangeIteratorCore : FeedIteratorInternal
+    {
+        internal readonly FeedRangeInternal FeedRangeInternal;
+        internal FeedRangeContinuation FeedRangeContinuation { get; private set; }        
+        private readonly ContainerCore containerCore;
+        private readonly CosmosClientContext clientContext;
+        private readonly QueryRequestOptions queryRequestOptions;
+        private readonly AsyncLazy<TryCatch<string>> lazyContainerRid;
+        private bool hasMoreResultsInternal;
+
+        public static FeedRangeIteratorCore Create(
+            ContainerCore containerCore,
+            FeedRangeInternal feedRangeInternal,
+            string continuation,
+            QueryRequestOptions options)
+        {
+            if (!string.IsNullOrEmpty(continuation))
+            {
+                if (FeedRangeContinuation.TryCreateFromString(continuation, out FeedRangeContinuation feedRangeContinuation))
```

If there is a version then we could avoid the ambiguity between bad continuation token that should throw an exception vs the older version.

---

That would imply we have a way right now to validate a ReadFeed continuation, I did not find anything in our current code base. Is there a validator available?